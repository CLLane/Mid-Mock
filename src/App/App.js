import React, { Component } from 'react';
import Container from '../Container/Container.js'
import Form from '../Form/Form.js'
import './App.css';

export default class App extends Component {
  constructor() {
    super();
    this.state ={
      purchases: []
    }
  }

  componentDidMount(){
    fetch('http://localhost:3001/api/v1/purchases')
    .then(response => response.json())
    .then(data => this.setState({ purchases: data}))
    .catch(error => error.message)

  }
 

    postToApi(card){
        const options = {
          method: 'POST',
          body: JSON.stringify({ ...card }),
          headers: {
            'Content-Type': 'application/json'
          }
      }
      fetch('http://localhost:3001/api/v1/purchases/', options).then(res => res.json).catch(err => err.message)
    }


  createCard = (e, card) => {
    console.log(e);
    e.preventDefault();
    let cardWithId = {...card, id: Date.now()}
    this.postToApi(cardWithId)
    this.setState({ purchases: [...this.state.purchases, cardWithId]})
  }

  render() {
    return (
      <div className="App">
        <header>
          <h1 className='app-title'>My Order History</h1>
          <div className='purchase-form'>
          <Form createCard={ this.createCard }/>
          </div>
        </header>
        <div className='purchase-container'>
            <Container  data={this.state.purchases} />
        </div>
      </div>
    );
  }
}


