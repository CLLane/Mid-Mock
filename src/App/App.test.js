import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App createCard={jest.fn()}/>)
  })

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })

//   it('should update state when createCard is called', () => {
//     const mockEvent = {target: {img: 'url', name: 'Chris', description: 'Has a small amount of confidence in this test...lol', price: 'to damn high'}}
//   })

//     const expectedName = {name: 'Chris'}

//     wrapper.instance().createCard(mockEvent)
})