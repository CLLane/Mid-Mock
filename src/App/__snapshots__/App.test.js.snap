// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapShot 1`] = `
<div
  className="App"
>
  <header>
    <h1
      className="app-title"
    >
      My Order History
    </h1>
    <div
      className="purchase-form"
    >
      <Form
        createCard={[Function]}
      />
    </div>
  </header>
  <div
    className="purchase-container"
  >
    <Container
      data={Array []}
    />
  </div>
</div>
`;
